### PR TITLE
feat(snapshot/rpc): add service transport rpc for create&get

### PR DESCRIPTION
**Pull request by tiagolobocastro**

### CODE REVIEW

File: control-plane/grpc/proto/v1/misc/common.proto [hunk 1]
```diff
@@ -110,61 +110,67 @@ enum ResourceKind {
...
 }
 
 // Filter by Node and Nexus id
-message NodeNexusFilter{
+message NodeNexusFilter {
   string node_id = 1;
   string nexus_id = 2;
 }
 
+// Filter by Snapshot id
```
Comment (Abhinandan-Purkait) [inline control-plane/grpc/proto/v1/misc/common.proto]: `Filter by Volume Id and Snapshot Id` ?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/grpc/src/operations/volume/traits_snapshots.rs [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+};
+
+use std::{collections::HashMap, convert::TryFrom};
+use stor_port::types::v0::{
+    store::SpecStatus,
+    transport::{SnapshotId, SnapshotTxId},
+};
+
+/// Volume snapshot information.
+pub trait ICreateVolumeSnapshot: Send + Sync + std::fmt::Debug {
```
Comment (Abhinandan-Purkait) [inline control-plane/grpc/src/operations/volume/traits_snapshots.rs]: Can we please rename this to something generic, then we can use the same for DeleteSnapshot as well. Or should we use a new one for that?

Comment (tiagolobocastro) [reply control-plane/grpc/src/operations/volume/traits_snapshots.rs]: yep, done.

File: control-plane/grpc/proto/v1/misc/common.proto (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/grpc/proto/v1/misc/common.proto]: Done

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**